### ISSUE TITLE
RHEL 7 - Unable to install SALT - Dependency errors

### ISSUE THREAD

**Issue by pcmccollum** [saltstack/salt#59366]
### Description of Issue
Unable to install SALT on RHEL 7 VM due to dependency errors:

### yum install:

```
Loaded plugins: product-id, search-disabled-repos, subscription-manager
Resolving Dependencies
--> Running transaction check
---> Package salt-master.noarch 0:3002.2-1.el8 will be installed
--> Processing Dependency: salt = 3002.2-1.el8 for package: salt-master-3002.2-1.el8.noarch
--> Processing Dependency: python3-systemd for package: salt-master-3002.2-1.el8.noarch
--> Running transaction check
---> Package salt.noarch 0:3002.2-1.el8 will be installed
--> Processing Dependency: python3-m2crypto >= 0.31.0 for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3-msgpack >= 0.4 for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3-zmq >= 17.0.0 for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3.6dist(distro) >= 1.2 for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3.6dist(m2crypto) >= 0.33.0 for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3.6dist(msgpack) >= 0.5 for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3.6dist(pyzmq) >= 17.0.0 for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3.6dist(requests) >= 1.0.0 for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3-distro for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3-jinja2 for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3-markupsafe for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3-psutil for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3-pycurl for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3-pyyaml for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3-requests for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3-rpm for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3-six for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3.6dist(jinja2) for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3.6dist(markupsafe) for package: salt-3002.2-1.el8.noarch
---> Package salt-master.noarch 0:3002.2-1.el8 will be installed
--> Processing Dependency: python3-systemd for package: salt-master-3002.2-1.el8.noarch
--> Running transaction check
---> Package python3-distro.noarch 0:1.2.0-4.el8 will be installed
---> Package python3-m2crypto.x86_64 0:0.33.0-1.el8 will be installed
--> Processing Dependency: libcrypto.so.1.1(OPENSSL_1_1_0)(64bit) for package: python3-m2crypto-0.33.0-1.el8.x86_64
--> Processing Dependency: libcrypto.so.1.1(OPENSSL_1_1_1)(64bit) for package: python3-m2crypto-0.33.0-1.el8.x86_64
--> Processing Dependency: libssl.so.1.1(OPENSSL_1_1_0)(64bit) for package: python3-m2crypto-0.33.0-1.el8.x86_64
--> Processing Dependency: python3-typing for package: python3-m2crypto-0.33.0-1.el8.x86_64
--> Processing Dependency: libcrypto.so.1.1()(64bit) for package: python3-m2crypto-0.33.0-1.el8.x86_64
--> Processing Dependency: libssl.so.1.1()(64bit) for package: python3-m2crypto-0.33.0-1.el8.x86_64
---> Package python3-msgpack.x86_64 0:0.6.1-3.el8 will be installed
---> Package python3-zmq.x86_64 0:17.0.0-5.el8 will be installed
--> Processing Dependency: libzmq.so.5()(64bit) for package: python3-zmq-17.0.0-5.el8.x86_64
---> Package python36-jinja2.noarch 0:2.11.1-1.el7 will be installed
---> Package python36-psutil.x86_64 0:5.6.7-1.el7 will be installed
---> Package python36-pycurl.x86_64 0:7.43.0-7.el7 will be installed
---> Package python36-requests.noarch 0:2.14.2-2.el7 will be installed
--> Processing Dependency: python36-chardet for package: python36-requests-2.14.2-2.el7.noarch
--> Processing Dependency: python36-idna for package: python36-requests-2.14.2-2.el7.noarch
--> Processing Dependency: python36-urllib3 for package: python36-requests-2.14.2-2.el7.noarch
---> Package python36-rpm.x86_64 0:4.11.3-9.el7 will be installed
---> Package salt.noarch 0:3002.2-1.el8 will be installed
--> Processing Dependency: python3-jinja2 for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3-markupsafe for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3-pyyaml for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3-six for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3.6dist(markupsafe) for package: salt-3002.2-1.el8.noarch
---> Package salt-master.noarch 0:3002.2-1.el8 will be installed
--> Processing Dependency: python3-systemd for package: salt-master-3002.2-1.el8.noarch
--> Running transaction check
---> Package openssl11-libs.x86_64 1:1.1.1g-2.el7 will be installed
---> Package python3-typing.noarch 0:3.5.2.2-4.el8 will be installed
---> Package python36-chardet.noarch 0:3.0.4-1.el7 will be installed
---> Package python36-idna.noarch 0:2.10-1.el7 will be installed
---> Package python36-urllib3.noarch 0:1.25.6-2.el7 will be installed
--> Processing Dependency: python36-six >= 1.12.0 for package: python36-urllib3-1.25.6-2.el7.noarch
--> Processing Dependency: python36-pysocks for package: python36-urllib3-1.25.6-2.el7.noarch
---> Package salt.noarch 0:3002.2-1.el8 will be installed
--> Processing Dependency: python3-jinja2 for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3-markupsafe for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3-pyyaml for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3-six for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3.6dist(markupsafe) for package: salt-3002.2-1.el8.noarch
---> Package salt-master.noarch 0:3002.2-1.el8 will be installed
--> Processing Dependency: python3-systemd for package: salt-master-3002.2-1.el8.noarch
---> Package zeromq.x86_64 0:4.3.1-4.el8 will be installed
--> Processing Dependency: libstdc++.so.6(CXXABI_1.3.9)(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Processing Dependency: libstdc++.so.6(GLIBCXX_3.4.20)(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Processing Dependency: libstdc++.so.6(GLIBCXX_3.4.21)(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Processing Dependency: libpgm-5.2.so.0()(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Processing Dependency: libsodium.so.23()(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Processing Dependency: libunwind.so.8()(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Running transaction check
---> Package libsodium.x86_64 0:1.0.18-1.el7 will be installed
---> Package libunwind.x86_64 2:1.2-2.el7 will be installed
---> Package openpgm.x86_64 0:5.2.122-17.el8 will be installed
---> Package python36-pysocks.noarch 0:1.6.8-7.el7 will be installed
---> Package python36-six.noarch 0:1.14.0-2.el7 will be installed
---> Package salt.noarch 0:3002.2-1.el8 will be installed
--> Processing Dependency: python3-jinja2 for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3-markupsafe for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3-pyyaml for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3-six for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3.6dist(markupsafe) for package: salt-3002.2-1.el8.noarch
---> Package salt-master.noarch 0:3002.2-1.el8 will be installed
--> Processing Dependency: python3-systemd for package: salt-master-3002.2-1.el8.noarch
---> Package zeromq.x86_64 0:4.3.1-4.el8 will be installed
--> Processing Dependency: libstdc++.so.6(CXXABI_1.3.9)(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Processing Dependency: libstdc++.so.6(GLIBCXX_3.4.20)(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Processing Dependency: libstdc++.so.6(GLIBCXX_3.4.21)(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Finished Dependency Resolution
Error: Package: salt-3002.2-1.el8.noarch (salt-py3-latest)
           Requires: python3.6dist(markupsafe)
Error: Package: salt-3002.2-1.el8.noarch (salt-py3-latest)
           Requires: python3-markupsafe
Error: Package: zeromq-4.3.1-4.el8.x86_64 (salt-py3-latest)
           Requires: libstdc++.so.6(GLIBCXX_3.4.21)(64bit)
Error: Package: salt-3002.2-1.el8.noarch (salt-py3-latest)
           Requires: python3-pyyaml
Error: Package: salt-3002.2-1.el8.noarch (salt-py3-latest)
           Requires: python3-six
Error: Package: salt-master-3002.2-1.el8.noarch (salt-py3-latest)
           Requires: python3-systemd
Error: Package: zeromq-4.3.1-4.el8.x86_64 (salt-py3-latest)
           Requires: libstdc++.so.6(GLIBCXX_3.4.20)(64bit)
Error: Package: salt-3002.2-1.el8.noarch (salt-py3-latest)
           Requires: python3-jinja2
Error: Package: zeromq-4.3.1-4.el8.x86_64 (salt-py3-latest)
           Requires: libstdc++.so.6(CXXABI_1.3.9)(64bit)
**********************************************************************
Dependency resolving failed due to missing dependencies.
Some repositories on your system are disabled, but yum can enable them
and search for missing dependencies. This will require downloading
metadata for disabled repositories and may take some time and traffic.
**********************************************************************

Enable all repositories and try again? [y/N]: y
https://cdn.redhat.com/content/eus/rhel/server/7/7Server/x86_64/sat-tools/6.8/os/repodata/repomd.xml: [Errno 14] HTTPS Error 404 - Not Found
Trying other mirror.
To address this issue please refer to the below knowledge base article

https://access.redhat.com/articles/1320623

If above article doesn't help to resolve this issue please open a ticket with Red Hat Support.

https://cdn.redhat.com/content/e4s/rhel/server/7/7Server/x86_64/sat-tools/6.2/os/repodata/repomd.xml: [Errno 14] HTTPS Error 404 - Not Found
Trying other mirror.
https://cdn.redhat.com/content/e4s/rhel/server/7/7Server/x86_64/sat-tools/6.1/os/repodata/repomd.xml: [Errno 14] HTTPS Error 404 - Not Found
Trying other mirror.
https://cdn.redhat.com/content/eus/rhel/server/7/7Server/x86_64/sat-tools/6.6/os/repodata/repomd.xml: [Errno 14] HTTPS Error 404 - Not Found
Trying other mirror.
https://cdn.redhat.com/content/eus/rhel/server/7/7Server/x86_64/sat-tools/6.7/os/repodata/repomd.xml: [Errno 14] HTTPS Error 404 - Not Found
Trying other mirror.
--> Running transaction check
---> Package salt.noarch 0:3002.2-1.el8 will be installed
--> Processing Dependency: python3-jinja2 for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3-markupsafe for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3-pyyaml for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3-six for package: salt-3002.2-1.el8.noarch
--> Processing Dependency: python3.6dist(markupsafe) for package: salt-3002.2-1.el8.noarch
---> Package salt-master.noarch 0:3002.2-1.el8 will be installed
--> Processing Dependency: python3-systemd for package: salt-master-3002.2-1.el8.noarch
---> Package zeromq.x86_64 0:4.3.1-4.el8 will be installed
--> Processing Dependency: libstdc++.so.6(CXXABI_1.3.9)(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Processing Dependency: libstdc++.so.6(GLIBCXX_3.4.20)(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Processing Dependency: libstdc++.so.6(GLIBCXX_3.4.21)(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Running transaction check
---> Package python3-jinja2.noarch 0:2.11.2-1.el7pc will be installed
---> Package python3-markupsafe.x86_64 0:1.1.1-2.el7pc will be obsoleting
---> Package python3-pyyaml.x86_64 0:5.3.1-3.el7pc will be obsoleting
---> Package python3-six.noarch 0:1.15.0-1.el7pc will be installed
---> Package python36-PyYAML.x86_64 0:3.13-1.el7 will be obsoleted
---> Package python36-markupsafe.x86_64 0:0.23-3.el7 will be obsoleted
---> Package salt-master.noarch 0:3002.2-1.el8 will be installed
--> Processing Dependency: python3-systemd for package: salt-master-3002.2-1.el8.noarch
---> Package zeromq.x86_64 0:4.3.1-4.el8 will be installed
--> Processing Dependency: libstdc++.so.6(CXXABI_1.3.9)(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Processing Dependency: libstdc++.so.6(GLIBCXX_3.4.20)(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Processing Dependency: libstdc++.so.6(GLIBCXX_3.4.21)(64bit) for package: zeromq-4.3.1-4.el8.x86_64
Removing python36-six.noarch 0:1.14.0-2.el7 - u due to obsoletes from python3-six.noarch 0:1.15.0-1.el7pc - u
Removing python36-jinja2.noarch 0:2.11.1-1.el7 - u due to obsoletes from python3-jinja2.noarch 0:2.11.2-1.el7pc - u
--> Restarting Dependency Resolution with new changes.
--> Running transaction check
---> Package python36-jinja2.noarch 0:2.11.1-1.el7 will be installed
---> Package python36-six.noarch 0:1.14.0-2.el7 will be installed
---> Package salt-master.noarch 0:3002.2-1.el8 will be installed
--> Processing Dependency: python3-systemd for package: salt-master-3002.2-1.el8.noarch
---> Package zeromq.x86_64 0:4.3.1-4.el8 will be installed
--> Processing Dependency: libstdc++.so.6(CXXABI_1.3.9)(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Processing Dependency: libstdc++.so.6(GLIBCXX_3.4.20)(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Processing Dependency: libstdc++.so.6(GLIBCXX_3.4.21)(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Finished Dependency Resolution
Error: Package: zeromq-4.3.1-4.el8.x86_64 (salt-py3-latest)
           Requires: libstdc++.so.6(GLIBCXX_3.4.21)(64bit)
Error: Package: salt-master-3002.2-1.el8.noarch (salt-py3-latest)
           Requires: python3-systemd
Error: Package: zeromq-4.3.1-4.el8.x86_64 (salt-py3-latest)
           Requires: libstdc++.so.6(GLIBCXX_3.4.20)(64bit)
Error: Package: zeromq-4.3.1-4.el8.x86_64 (salt-py3-latest)
           Requires: libstdc++.so.6(CXXABI_1.3.9)(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
```

### Script Install:
```
 *  INFO: Running version: 2020.10.20
 *  INFO: Executed by: -
 *  INFO: Command line: './install_salt.sh -M git develop'

 *  INFO: System Information:
 *  INFO:   CPU:          GenuineIntel
 *  INFO:   CPU Arch:     x86_64
 *  INFO:   OS Name:      Linux
 *  INFO:   OS Version:   3.10.0-957.el7.x86_64
 *  INFO:   Distribution: <R>ed <H>at <E>nterprise <L>inux 7.9

 *  INFO: Installing minion
 *  INFO: Installing master
 *  INFO: Found function install_red_hat_enterprise_linux_git_deps
 *  INFO: Found function config_salt
 *  INFO: Found function preseed_master
 *  INFO: Found function install_red_hat_enterprise_linux_git
 *  INFO: Found function install_red_hat_enterprise_linux_git_post
 *  INFO: Found function install_red_hat_enterprise_linux_restart_daemons
 *  INFO: Found function daemons_running
 *  INFO: Running install_red_hat_enterprise_linux_git_deps()
Loaded plugins: product-id, search-disabled-repos, subscription-manager
Package yum-utils-1.1.31-54.el7_8.noarch already installed and latest version
Package chkconfig-1.7.6-1.el7.x86_64 already installed and latest version
Package PyYAML-3.11-1.el7.x86_64 already installed and latest version
Nothing to do
Loaded plugins: product-id, search-disabled-repos, subscription-manager
Package ca-certificates-2020.2.41-70.0.el7_8.noarch already installed and latest version
Nothing to do
 *  WARN: The git revision being installed does not match a Salt version tag. Shallow cloning disabled
Cloning into 'salt'...
Switched to a new branch 'develop'
Branch develop set up to track remote branch develop from origin.
 *  INFO: Cloning Salt's git repository succeeded
Loaded plugins: product-id, search-disabled-repos, subscription-manager
Package python-2.7.5-90.el7.x86_64 already installed and latest version
Package python-futures is obsoleted by python2-futures, trying to install python2-futures-3.1.1-5.el7.noarch instead
Package python-crypto is obsoleted by python2-crypto, trying to install python2-crypto-2.6.1-16.el7.x86_64 instead
Package python-jinja2-2.7.2-4.el7.noarch already installed and latest version
No package python-tornado available.
Resolving Dependencies
--> Running transaction check
---> Package m2crypto.x86_64 0:0.21.1-17.el7 will be updated
---> Package m2crypto.x86_64 0:0.31.0-4.el7 will be an update
--> Processing Dependency: python2-typing for package: m2crypto-0.31.0-4.el7.x86_64
---> Package python-requests.noarch 0:2.6.0-10.el7 will be installed
--> Processing Dependency: python-urllib3 >= 1.10.2-1 for package: python-requests-2.6.0-10.el7.noarch
---> Package python-zmq.x86_64 0:15.3.0-3.el7 will be installed
--> Processing Dependency: zeromq >= 4 for package: python-zmq-15.3.0-3.el7.x86_64
--> Processing Dependency: libzmq.so.5()(64bit) for package: python-zmq-15.3.0-3.el7.x86_64
---> Package python2-crypto.x86_64 0:2.6.1-16.el7 will be installed
--> Processing Dependency: libtomcrypt.so.0()(64bit) for package: python2-crypto-2.6.1-16.el7.x86_64
---> Package python2-futures.noarch 0:3.1.1-5.el7 will be installed
---> Package python2-msgpack.x86_64 0:0.6.2-2.el7 will be installed
---> Package systemd-python.x86_64 0:219-78.el7_9.2 will be installed
--> Running transaction check
---> Package libtomcrypt.x86_64 0:1.17-25.el7 will be installed
--> Processing Dependency: libtommath >= 0.42.0 for package: libtomcrypt-1.17-25.el7.x86_64
--> Processing Dependency: libtommath.so.0()(64bit) for package: libtomcrypt-1.17-25.el7.x86_64
---> Package python-urllib3.noarch 0:1.10.2-7.el7 will be installed
---> Package python2-typing.noarch 0:3.5.2.2-4.el7 will be installed
---> Package zeromq.x86_64 0:4.3.1-4.el8 will be installed
--> Processing Dependency: libstdc++.so.6(CXXABI_1.3.9)(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Processing Dependency: libstdc++.so.6(GLIBCXX_3.4.20)(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Processing Dependency: libstdc++.so.6(GLIBCXX_3.4.21)(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Processing Dependency: libpgm-5.2.so.0()(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Processing Dependency: libsodium.so.23()(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Processing Dependency: libunwind.so.8()(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Running transaction check
---> Package libsodium.x86_64 0:1.0.18-1.el7 will be installed
---> Package libtommath.x86_64 0:0.42.0-5.el7 will be installed
---> Package libunwind.x86_64 2:1.2-2.el7 will be installed
---> Package openpgm.x86_64 0:5.2.122-17.el8 will be installed
---> Package zeromq.x86_64 0:4.3.1-4.el8 will be installed
--> Processing Dependency: libstdc++.so.6(CXXABI_1.3.9)(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Processing Dependency: libstdc++.so.6(GLIBCXX_3.4.20)(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Processing Dependency: libstdc++.so.6(GLIBCXX_3.4.21)(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Finished Dependency Resolution
Error: Package: zeromq-4.3.1-4.el8.x86_64 (salt-py3-latest)
           Requires: libstdc++.so.6(GLIBCXX_3.4.21)(64bit)
Error: Package: zeromq-4.3.1-4.el8.x86_64 (salt-py3-latest)
           Requires: libstdc++.so.6(GLIBCXX_3.4.20)(64bit)
Error: Package: zeromq-4.3.1-4.el8.x86_64 (salt-py3-latest)
           Requires: libstdc++.so.6(CXXABI_1.3.9)(64bit)
**********************************************************************
Dependency resolving failed due to missing dependencies.
Some repositories on your system are disabled, but yum can enable them
and search for missing dependencies. This will require downloading
metadata for disabled repositories and may take some time and traffic.
**********************************************************************

https://cdn.redhat.com/content/eus/rhel/server/7/7Server/x86_64/sat-tools/6.8/os/repodata/repomd.xml: [Errno 14] HTTPS Error 404 - Not Found
Trying other mirror.
To address this issue please refer to the below knowledge base article

https://access.redhat.com/articles/1320623

If above article doesn't help to resolve this issue please open a ticket with Red Hat Support.

https://cdn.redhat.com/content/e4s/rhel/server/7/7Server/x86_64/sat-tools/6.2/os/repodata/repomd.xml: [Errno 14] HTTPS Error 404 - Not Found
Trying other mirror.
https://cdn.redhat.com/content/e4s/rhel/server/7/7Server/x86_64/sat-tools/6.1/os/repodata/repomd.xml: [Errno 14] HTTPS Error 404 - Not Found
Trying other mirror.
https://cdn.redhat.com/content/eus/rhel/server/7/7Server/x86_64/sat-tools/6.6/os/repodata/repomd.xml: [Errno 14] HTTPS Error 404 - Not Found
Trying other mirror.
https://cdn.redhat.com/content/eus/rhel/server/7/7Server/x86_64/sat-tools/6.7/os/repodata/repomd.xml: [Errno 14] HTTPS Error 404 - Not Found
Trying other mirror.
--> Running transaction check
---> Package zeromq.x86_64 0:4.3.1-4.el8 will be installed
--> Processing Dependency: libstdc++.so.6(CXXABI_1.3.9)(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Processing Dependency: libstdc++.so.6(GLIBCXX_3.4.20)(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Processing Dependency: libstdc++.so.6(GLIBCXX_3.4.21)(64bit) for package: zeromq-4.3.1-4.el8.x86_64
--> Finished Dependency Resolution
Error: Package: zeromq-4.3.1-4.el8.x86_64 (salt-py3-latest)
           Requires: libstdc++.so.6(GLIBCXX_3.4.21)(64bit)
Error: Package: zeromq-4.3.1-4.el8.x86_64 (salt-py3-latest)
           Requires: libstdc++.so.6(GLIBCXX_3.4.20)(64bit)
Error: Package: zeromq-4.3.1-4.el8.x86_64 (salt-py3-latest)
           Requires: libstdc++.so.6(CXXABI_1.3.9)(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
 * ERROR: Failed to run install_red_hat_enterprise_linux_git_deps()!!!
```

### Setup
RHEL 7 VM

### Steps to Reproduce Issue
I've tried installing both via yum:

`yum install salt-master`

and the SALT install shell script:

`./install_salt.sh -M git develop`

### Versions Report
N/A

**Comment by sagetherage** [comment]
@pcmccollum there are many reasons a `yum install` will give dependency errors so I will assign this to help there, but you are also trying to install from `develop` and we no longer maintain `develop` as our main branch - have you tried from `master`?

**Comment by pcmccollum** [comment]
@sagetherage Thank you for the support. I did try `master` and it seemed to have a different set of errors:

```
 *  INFO: Running version: 2020.10.20
 *  INFO: Executed by: -
 *  INFO: Command line: './install_salt.sh -M git master'

 *  INFO: System Information:
 *  INFO:   CPU:          GenuineIntel
 *  INFO:   CPU Arch:     x86_64
 *  INFO:   OS Name:      Linux
 *  INFO:   OS Version:   3.10.0-957.el7.x86_64
 *  INFO:   Distribution: <R>ed <H>at <E>nterprise <L>inux 7.9

 *  INFO: Installing minion
 *  INFO: Installing master

 *  WARN: Post Neon git based installations will always install salt
 *  WARN: and its dependencies using pip which will be upgraded to
 *  WARN: at least v8.0.0, and, in case the setuptools version is also
 *  WARN: too old, it will be upgraded to at least v9.1

 *  WARN: You have 10 seconds to cancel and stop the bootstrap process...

 *  INFO: Found function install_red_hat_enterprise_linux_git_deps
 *  INFO: Found function config_salt
 *  INFO: Found function preseed_master
 *  INFO: Found function install_red_hat_enterprise_linux_git
 *  INFO: Found function install_red_hat_enterprise_linux_git_post
 *  INFO: Found function install_red_hat_enterprise_linux_restart_daemons
 *  INFO: Found function daemons_running
 *  INFO: Running install_red_hat_enterprise_linux_git_deps()
Loaded plugins: product-id, search-disabled-repos, subscription-manager
Package yum-utils-1.1.31-54.el7_8.noarch already installed and latest version
Package chkconfig-1.7.6-1.el7.x86_64 already installed and latest version
Nothing to do
Loaded plugins: product-id, search-disabled-repos, subscription-manager
Package ca-certificates-2020.2.41-70.0.el7_8.noarch already installed and latest version
Nothing to do
 *  INFO: Git revision matches a Salt version tag, shallow cloning enabled.
 *  INFO: Attempting to shallow clone master from Salt's repository https://github.com/saltstack/salt.git
Cloning into 'salt'...
 *  INFO: Cloning Salt's git repository succeeded
Loaded plugins: product-id, search-disabled-repos, subscription-manager
Package python-2.7.5-90.el7.x86_64 already installed and latest version
Package python2-pip-8.1.2-14.el7.noarch already installed and latest version
Resolving Dependencies
--> Running transaction check
---> Package gcc.x86_64 0:4.8.5-44.el7 will be installed
--> Processing Dependency: cpp = 4.8.5-44.el7 for package: gcc-4.8.5-44.el7.x86_64
--> Processing Dependency: glibc-devel >= 2.2.90-12 for package: gcc-4.8.5-44.el7.x86_64
--> Processing Dependency: libmpc.so.3()(64bit) for package: gcc-4.8.5-44.el7.x86_64
--> Processing Dependency: libmpfr.so.4()(64bit) for package: gcc-4.8.5-44.el7.x86_64
---> Package python-devel.x86_64 0:2.7.5-90.el7 will be installed
--> Processing Dependency: python-rpm-macros > 3-30 for package: python-devel-2.7.5-90.el7.x86_64
--> Processing Dependency: python2-rpm-macros > 3-30 for package: python-devel-2.7.5-90.el7.x86_64
---> Package python-setuptools.noarch 0:0.9.8-7.el7 will be updated
---> Package python-setuptools.noarch 0:36.6.0-3.ius.el7 will be an update
--> Running transaction check
---> Package cpp.x86_64 0:4.8.5-44.el7 will be installed
---> Package glibc-devel.x86_64 0:2.17-317.el7 will be installed
--> Processing Dependency: glibc-headers = 2.17-317.el7 for package: glibc-devel-2.17-317.el7.x86_64
--> Processing Dependency: glibc-headers for package: glibc-devel-2.17-317.el7.x86_64
---> Package libmpc.x86_64 0:1.0.1-3.el7 will be installed
---> Package mpfr.x86_64 0:3.1.1-4.el7 will be installed
---> Package python-rpm-macros.noarch 0:3-34.el7 will be installed
--> Processing Dependency: python-srpm-macros for package: python-rpm-macros-3-34.el7.noarch
---> Package python2-rpm-macros.noarch 0:3-34.el7 will be installed
--> Running transaction check
---> Package glibc-headers.x86_64 0:2.17-317.el7 will be installed
--> Processing Dependency: kernel-headers >= 2.2.1 for package: glibc-headers-2.17-317.el7.x86_64
--> Processing Dependency: kernel-headers for package: glibc-headers-2.17-317.el7.x86_64
---> Package python-srpm-macros.noarch 0:3-34.el7 will be installed
--> Running transaction check
---> Package kernel-headers.x86_64 0:3.10.0-1160.11.1.el7 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

================================================================================
 Package             Arch    Version                  Repository           Size
================================================================================
Installing:
 gcc                 x86_64  4.8.5-44.el7             rhel-7-server-rpms   16 M
 python-devel        x86_64  2.7.5-90.el7             rhel-7-server-rpms  399 k
Updating:
 python-setuptools   noarch  36.6.0-3.ius.el7         saltstack           560 k
Installing for dependencies:
 cpp                 x86_64  4.8.5-44.el7             rhel-7-server-rpms  6.0 M
 glibc-devel         x86_64  2.17-317.el7             rhel-7-server-rpms  1.1 M
 glibc-headers       x86_64  2.17-317.el7             rhel-7-server-rpms  690 k
 kernel-headers      x86_64  3.10.0-1160.11.1.el7     rhel-7-server-rpms  9.0 M
 libmpc              x86_64  1.0.1-3.el7              rhel-7-server-rpms   51 k
 mpfr                x86_64  3.1.1-4.el7              rhel-7-server-rpms  203 k
 python-rpm-macros   noarch  3-34.el7                 rhel-7-server-rpms  9.1 k
 python-srpm-macros  noarch  3-34.el7                 rhel-7-server-rpms  8.8 k
 python2-rpm-macros  noarch  3-34.el7                 rhel-7-server-rpms  8.1 k

Transaction Summary
================================================================================
Install  2 Packages (+9 Dependent packages)
Upgrade  1 Package

Total download size: 34 M
Downloading packages:
Delta RPMs disabled because /usr/bin/applydeltarpm not installed.
--------------------------------------------------------------------------------
Total                                              8.6 MB/s |  34 MB  00:03
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Installing : mpfr-3.1.1-4.el7.x86_64                                     1/13
  Installing : libmpc-1.0.1-3.el7.x86_64                                   2/13
  Installing : cpp-4.8.5-44.el7.x86_64                                     3/13
  Installing : python-srpm-macros-3-34.el7.noarch                          4/13
  Installing : python-rpm-macros-3-34.el7.noarch                           5/13
  Installing : kernel-headers-3.10.0-1160.11.1.el7.x86_64                  6/13
  Installing : glibc-headers-2.17-317.el7.x86_64                           7/13
  Installing : glibc-devel-2.17-317.el7.x86_64                             8/13
  Installing : python2-rpm-macros-3-34.el7.noarch                          9/13
  Installing : python-devel-2.7.5-90.el7.x86_64                           10/13
  Installing : gcc-4.8.5-44.el7.x86_64                                    11/13
  Updating   : python-setuptools-36.6.0-3.ius.el7.noarch                  12/13
  Cleanup    : python-setuptools-0.9.8-7.el7.noarch                       13/13
  Verifying  : python2-rpm-macros-3-34.el7.noarch                          1/13
  Verifying  : python-rpm-macros-3-34.el7.noarch                           2/13
  Verifying  : python-devel-2.7.5-90.el7.x86_64                            3/13
  Verifying  : python-setuptools-36.6.0-3.ius.el7.noarch                   4/13
  Verifying  : kernel-headers-3.10.0-1160.11.1.el7.x86_64                  5/13
  Verifying  : mpfr-3.1.1-4.el7.x86_64                                     6/13
  Verifying  : glibc-headers-2.17-317.el7.x86_64                           7/13
  Verifying  : cpp-4.8.5-44.el7.x86_64                                     8/13
  Verifying  : glibc-devel-2.17-317.el7.x86_64                             9/13
  Verifying  : gcc-4.8.5-44.el7.x86_64                                    10/13
  Verifying  : libmpc-1.0.1-3.el7.x86_64                                  11/13
  Verifying  : python-srpm-macros-3-34.el7.noarch                         12/13
  Verifying  : python-setuptools-0.9.8-7.el7.noarch                       13/13

Installed:
  gcc.x86_64 0:4.8.5-44.el7          python-devel.x86_64 0:2.7.5-90.el7

Dependency Installed:
  cpp.x86_64 0:4.8.5-44.el7
  glibc-devel.x86_64 0:2.17-317.el7
  glibc-headers.x86_64 0:2.17-317.el7
  kernel-headers.x86_64 0:3.10.0-1160.11.1.el7
  libmpc.x86_64 0:1.0.1-3.el7
  mpfr.x86_64 0:3.1.1-4.el7
  python-rpm-macros.noarch 0:3-34.el7
  python-srpm-macros.noarch 0:3-34.el7
  python2-rpm-macros.noarch 0:3-34.el7

Updated:
  python-setuptools.noarch 0:36.6.0-3.ius.el7

Complete!
 *  INFO: Running config_salt()
 *  INFO: Running install_red_hat_enterprise_linux_git()
Traceback (most recent call last):
  File "/bin/pip2.7", line 9, in <module>
    load_entry_point('pip==21.0', 'console_scripts', 'pip2.7')()
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 570, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2755, in load_entry_point
    return ep.load()
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2409, in load
    return self.resolve()
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2415, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/lib/python2.7/site-packages/pip/_internal/cli/main.py", line 60
    sys.stderr.write(f"ERROR: {exc}")
                                   ^
SyntaxError: invalid syntax
Desired pip version '8.0.0' < Installed pip version '21.0'
Traceback (most recent call last):
  File "/bin/pip2.7", line 9, in <module>
    load_entry_point('pip==21.0', 'console_scripts', 'pip2.7')()
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 570, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2755, in load_entry_point
    return ep.load()
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2409, in load
    return self.resolve()
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2415, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/lib/python2.7/site-packages/pip/_internal/cli/main.py", line 60
    sys.stderr.write(f"ERROR: {exc}")
                                   ^
SyntaxError: invalid syntax
 *  INFO: Installing salt using python
 *  INFO: Downloading Salt Dependencies from PyPi
Traceback (most recent call last):
  File "/bin/pip2.7", line 9, in <module>
    load_entry_point('pip==21.0', 'console_scripts', 'pip2.7')()
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 570, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2755, in load_entry_point
    return ep.load()
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2409, in load
    return self.resolve()
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2415, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/lib/python2.7/site-packages/pip/_internal/cli/main.py", line 60
    sys.stderr.write(f"ERROR: {exc}")
                                   ^
SyntaxError: invalid syntax
Failed to download salt dependencies
 *  INFO: Installing Downloaded Salt Dependencies
Traceback (most recent call last):
  File "/bin/pip2.7", line 9, in <module>
    load_entry_point('pip==21.0', 'console_scripts', 'pip2.7')()
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 570, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2755, in load_entry_point
    return ep.load()
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2409, in load
    return self.resolve()
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2415, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/lib/python2.7/site-packages/pip/_internal/cli/main.py", line 60
    sys.stderr.write(f"ERROR: {exc}")
                                   ^
SyntaxError: invalid syntax
 * ERROR: Failed to run install_red_hat_enterprise_linux_git()!!!
```

**Comment by sagetherage** [comment]
@krionbsd let me know if you can get this triaged this week or not, I can attempt to reassign if you cannot, thank you!

**Comment by danielrobbins** [comment]
@pcmccollum I can't reproduce on CentOS 7 in a container. I believe some of the packages that are being installed require an updated libstdc++ to be installed.

Can you please:

1. Include the exact command(s) you are using to install salt-master.
2. Include the exact version of RHEL 7 you are using.
3. Use yum to update your system and see if this resolves the issue.

**Comment by sagetherage** [comment]
@pcmccollum following up here - since it has been some time have you been able to upgrade to a different RHEL version and/or salt version?

**Comment by ConstableBall** [comment]
I am facing same issue upon installing salt minion on centos 7.9.

--> Processing Dependency: python3-jinja2 for package: salt-3003.1-1.el8.noarch
--> Processing Dependency: python3-markupsafe for package: salt-3003.1-1.el8.noarch
--> Processing Dependency: python3-pyyaml for package: salt-3003.1-1.el8.noarch
--> Processing Dependency: python3.6dist(markupsafe) for package: salt-3003.1-1.el8.noarch
---> Package zeromq.x86_64 0:4.3.4-2.el8 will be installed
--> Processing Dependency: libstdc++.so.6(CXXABI_1.3.9)(64bit) for package: zeromq-4.3.4-2.el8.x86_64
--> Processing Dependency: libstdc++.so.6(GLIBCXX_3.4.20)(64bit) for package: zeromq-4.3.4-2.el8.x86_64
--> Processing Dependency: libstdc++.so.6(GLIBCXX_3.4.21)(64bit) for package: zeromq-4.3.4-2.el8.x86_64
--> Finished Dependency Resolution
Error: Package: zeromq-4.3.4-2.el8.x86_64 (salt-latest-repo)
           Requires: libstdc++.so.6(CXXABI_1.3.9)(64bit)
Error: Package: salt-3003.1-1.el8.noarch (salt-latest-repo)
           Requires: python3-pyyaml
Error: Package: libsodium-1.0.18-2.el8.x86_64 (salt-latest-repo)
           Requires: libc.so.6(GLIBC_2.25)(64bit)
Error: Package: salt-3003.1-1.el8.noarch (salt-latest-repo)
           Requires: python3.6dist(markupsafe)
Error: Package: salt-3003.1-1.el8.noarch (salt-latest-repo)
           Requires: python3-markupsafe
Error: Package: salt-3003.1-1.el8.noarch (salt-latest-repo)
           Requires: python3-jinja2
Error: Package: zeromq-4.3.4-2.el8.x86_64 (salt-latest-repo)
           Requires: libstdc++.so.6(GLIBCXX_3.4.20)(64bit)
Error: Package: zeromq-4.3.4-2.el8.x86_64 (salt-latest-repo)
           Requires: libstdc++.so.6(GLIBCXX_3.4.21)(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
 * ERROR: Failed to run install_centos_stable()!!!
[root@localhost ~]# cat /etc/centos-release
CentOS Linux release 7.9.2009 (Core)

step to reproduce:
url -fsSL https://bootstrap.saltproject.io -o install_salt.sh
sudo sh install_salt.sh -P -x python3

**Comment by sagetherage** [comment]
> centos 7.9.

@ConstableBall and you can confirm your system up-to-date with the latest Centos 7.9 packages? and you are running the latest Salt packages as well?

**Comment by ScriptAutomate** [comment]
I'm unable to reproduce this locally on CentOS 7.9 (I dropped into a docker container):

```bash
docker container run --rm -it centos:7 /bin/bash
```

```bash
# Install salt via bootstrap
curl -fsSL https://bootstrap.saltproject.io -o install_salt.sh
sudo sh install_salt.sh -P -x python3
```

Version info output:

```bash
[root@2fa5e7d13f29 /]# salt-call --local grains.item os osrelease saltversion
local:
    ----------
    os:
        CentOS
    osrelease:
        7.9.2009
    saltversion:
        3003.1
```

I tried also via `git master` as @pcmccollum did at the start (using `master` due to how we shouldn't use `develop` anymore):

```bash
curl -fsSL https://bootstrap.saltproject.io -o install_salt.sh
sudo sh install_salt.sh -M git master
```

This is **a bad idea as it is Python 2 by default if not using the `-x python3` arg.** It fails for me. Going with Python 3 fixes this, which is:

```bash
curl -fsSL https://bootstrap.saltproject.io -o install_salt.sh
sudo sh install_salt.sh -x python3 -M git master
```

This _does_ work for me. We generally suggest **not to go with a direct call to `git master` as that is technically an experimental build and not a release.**

Fix should be, if wanting to use `git master`:

- Don't use this: `sudo sh install_salt.sh -M git master`
- Use this! `sudo sh install_salt.sh -x python3 -M git master`

REMINDER: Salt v3000.x transitions to Extended Life Support after August 2021 (end of this month). What does that mean?

> **Extended life support**
During extended life support, Salt Project will not release any point releases. Users will still have access to any documentation. Salt Project will provide ongoing best-effort technical support for customers on existing installations. No bug fixes, security fixes, improved functionality, or root-cause analysis will be provided.

To stay up-to-date, upgrade! **This is especially the case for all users of the Python 2 version of Salt v3000.x, as it is the last major version to support Python 2 (also EOL).**
> More information on the Salt lifecycle can be seen here: https://saltproject.io/salt-platform-support/
> A blog post will be coming out with additional information next week, and we will remind the community again in Slack, the Open Hour, and our other communication channels.

@ConstableBall I'm not sure what is going on with yours, as your errors are showing **el8** (instead of **el7** / CentOS 7) related errors. I have tried the same bootstrap against both CentOS 7 and CentOS 8, and I am unable to reproduce. Can you try again, and also provide the output for:

```bash
cat /etc/*ease*
```

**Comment by ScriptAutomate** [comment]
> @ConstableBall I'm not sure what is going on with yours, as your errors are showing **el8** (instead of **el7** / CentOS 7) related errors. I have tried the same bootstrap against both CentOS 7 and CentOS 8, and I am unable to reproduce. Can you try again, and also provide the output for:
> 
> ```shell
> cat /etc/*ease*
> ```

I'm going to close this issue, but a new issue can be created if this is still a problem for @ConstableBall. If it is still an issue, open a new issue with the same details, but also provide the detailed output of `cat /etc/*ease*` on the system, along with verifying that these commands have been run on CentOS 7 (as your output suggests CentOS 8, or incorrect functionality coming from `salt-bootstrap`).

If a new issue is raised, it should be raised in:
- https://github.com/saltstack/salt-bootstrap

As it may be specifically bootstrap related. To confirm whether Salt doesn't install at all, regardless of using bootstrap, attempt installing the direct route:

```bash
# Directions from https://repo.saltproject.io/#rhel
sudo rpm --import https://repo.saltproject.io/py3/redhat/7/x86_64/latest/SALTSTACK-GPG-KEY.pub
curl -fsSL https://repo.saltproject.io/py3/redhat/7/x86_64/latest.repo | sudo tee /etc/yum.repos.d/salt.repo
```

**Comment by AlexdeTaranto** [comment]
I had exactly the same issue just now, trying to install Salt on Centos7.9. I hadn't spotted that I was trying to install using the instructions for RHEL/Centos8. As soon as I switched my salt repository to using the following repo data, it worked like a charm:

```
[saltstack-repo]
name=Salt repo for Red Hat Enterprise Linux $releasever
baseurl=https://repo.saltproject.io/py3/redhat/$releasever/$basearch/latest
enabled=1
gpgcheck=1
gpgkey=https://repo.saltproject.io/py3/redhat/$releasever/$basearch/latest/SALTSTACK-GPG-KEY.pub
       https://repo.saltproject.io/py3/redhat/$releasever/$basearch/latest/base/RPM-GPG-KEY-CentOS-7
```
